### PR TITLE
Changes from release feedback

### PR DESCRIPTION
**Pull request by darkma773r**
- Use `0` instead of `0.0` in if condition comparison
- Rename boolean getter methods to follow `isXXX` convention
- Add test for `AbstractNode.depth()` for detached nodes
- Remove inline checkstyle comments
- Fix user guide typos
- Use `String.toLowerCase(Locale.ROOT)` in `BoundaryIOManager`
- Use explicit `long` to `int` cast
- Remove unnecessary `if` condition
- Add javadoc external links
- Fix javadoc reference 
- Update PMD config

### CODE REVIEW

File: src/main/resources/pmd/pmd-ruleset.xml
Comment (aherbert) [inline src/main/resources/pmd/pmd-ruleset.xml:44]: Is it the intention to put this back in the future?

**Review by aherbert — APPROVED** [review]
All changes look good. There is an issue with the commons numbers javadocs on the live site. We can discuss on the mailing list.

File: src/main/resources/pmd/pmd-ruleset.xml
Comment (darkma773r) [reply src/main/resources/pmd/pmd-ruleset.xml:44]: No. I disagree with this in some cases since I often find it helpful to have frequently used inner types (particularly public enums and interfaces) at the top of the class.

**Comment by darkma773r** [comment]
Thanks, @aherbert. Are we good to go for another RC after I merge this?

**Comment by aherbert** [comment]
Yes. I think the issue with javadoc links to numbers is now a problem with the numbers site for the javadoc directories.

**Pull request merged**
